code/datums/traits/_quirk.dm

```diff
@@ -64,9 +58,19 @@
 /datum/quirk/proc/clone_data() //return additional data that should be remembered by cloning
 /datum/quirk/proc/on_clone(data) //create the quirk from clone data
 
+/datum/quirk/proc/clear_refs()
+	if(quirk_holder)
+		remove()
+		UnregisterSignal(quirk_holder, COMSIG_PARENT_QDELETING)
+		to_chat(quirk_holder, lose_text)
+		quirk_holder.roundstart_quirks -= src
+		if(mob_trait)
+			REMOVE_TRAIT(quirk_holder, mob_trait, ROUNDSTART_TRAIT)
+		quirk_holder = null
+
 /datum/quirk/proc/handle_parent_del()
 	SIGNAL_HANDLER
-	quirk_holder = null
+	clear_refs()
 	qdel(src)
```

Comment:
This would have worked before if quirk_holder = null was simply removed since removing it was handled in qdel(src), by setting quirk_holder to null before calling qdel causes the remove behaviour to not be executed.
This would fix it, but it could be fixed in 1 line too.